libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1054,15 +1056,17 @@ bool AP_GPS::get_lag(uint8_t instance, float &lag_sec) const
         // the user is always right !!
         return true;
     } else if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
-        // no GPS was detected in this instance so return a default delay of 1 measurement interval
-        lag_sec = 0.001f * (float)get_rate_ms(instance);
-        // check lack of GPS is consistent with user expectation
-        return state[instance].status == NO_GPS;
+        // no GPS was detected in this instance so return the worst possible lag term
+        if (_type[instance] == GPS_TYPE_NONE) {
+            lag_sec = 0.0f;
+            return true;
+        } else {
+            lag_sec = GPS_WORST_LAG_SEC;
+        }
+        return _type[instance] != GPS_TYPE_AUTO;
```

Comment:
I'm not sure I understand this line. Does this mean that when the user specifies a GPS type but it hasn't been detected yet, it will return the worst lag as valid, but when auto is selected it returns as not valid? It seems the contrary to what I would expect - when a GPS type is selected a GPS is expected to be there so a driver is expected to be created at some point, but with auto there may or may not exist a GPS.